telemetry-core/src/main/java/com/newrelic/telemetry/transport/BatchDataSender.java

```diff
@@ -71,24 +71,25 @@ private String buildUserAgent(String additionalUserAgent) {
     return BASE_USER_AGENT_VALUE + " " + additionalUserAgent;
   }
 
-  public Response send(String json, UUID batchId)
+  public Response send(String json, UUID batchId, String batchType)
```

Comment:
I would probably refactor the method signature here to `public Response send(String json, TelemetryBatch<? extends Telemetry> batch)` to just take a `TelemetryBatch` since you're getting both the `batchId` and `batchType` from the same `TelemetryBatch` object: `batch.getUuid()` and `batch.getClass().getSimpleName()`, respectively.

Reply:
I changed the signature.